compiler/rustc_target/src/spec/mod.rs

```diff
@@ -2446,8 +2477,15 @@ pub struct TargetOptions {
     pub llvm_mcount_intrinsic: Option<StaticCow<str>>,
 
     /// LLVM ABI name, corresponds to the '-mabi' parameter available in multilib C compilers
+    /// and the `-target-abi` flag in llc. In the LLVM API this is `MCOptions.ABIName`.
     pub llvm_abiname: StaticCow<str>,
 
+    /// Corresponds to `-mfloat-abi` parameter in clang/GCC and the `-float-abi` flag in llc.
```

Comment:
Not strictly true, same as the `softfp` comment above - `-mfloat-abi` in clang also has the special `softfp` value. Maybe rather than expanding the error message like I suggested, we could just drop a note in the docs about how to encode that?